huaweicloud/services/hss/resource_huaweicloud_hss_webtamper_protection.go

```diff
@@ -0,0 +1,336 @@
+// @API HSS GET /v5/{project_id}/host-management/hosts
+func ResourceWebTamperProtection() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceWebTamperProtectionCreate,
+		ReadContext:   resourceWebTamperProtectionRead,
+		UpdateContext: resourceWebTamperProtectionUpdate,
+		DeleteContext: resourceWebTamperProtectionDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceWebTamperProtectionImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"host_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"charging_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"quota_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"is_dynamics_protect": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"host_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"private_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"group_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os_bit": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protect_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rasp_protect_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"anti_tampering_times": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"detect_tampering_times": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func openWebTamperProtection(client *hssv5.HssClient, cfg *config.Config, hostId string, d *schema.ResourceData) error {
+	var (
+		region = cfg.GetRegion(d)
+		epsId  = cfg.GetEnterpriseProjectID(d)
+	)
+
+	requestBody := hssv5model.SetWtpProtectionStatusRequestInfo{
+		Status:     utils.Bool(true),
+		HostIdList: &[]string{hostId},
+	}
+	// The `quota_id` parameter needs to be set only when the charging mode of the quota is prePaid.
+	// Using postPaid quota, there is no need to set parameters `charging_mode` and `quota_id` when requesting APIs.
+	if v, ok := d.GetOk("charging_mode"); ok && v.(string) == chargingModePrePaid {
+		requestBody.ChargingMode = utils.String(chargingModePacketCycle)
+		requestBody.ResourceId = utils.StringIgnoreEmpty(d.Get("quota_id").(string))
+	}
+	opts := hssv5model.SetWtpProtectionStatusInfoRequest{
+		Region:              region,
+		EnterpriseProjectId: &epsId,
+		Body:                &requestBody,
+	}
+
+	_, err := client.SetWtpProtectionStatusInfo(&opts)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
```suggestion
	return err
```
If the error is not formatted, returned directly just ok.
other places ditto

Reply:
done.